charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -62,16 +62,30 @@ Create the name of the service account to use
 {{- end }}
 
 {{/*
-Build base config file with pipelines
 */}}
-{{- define "opentelemetry-collector.pipelinesConfig" -}}
-{{- $config := include "opentelemetry-collector.baseConfig" . | fromYaml -}}
-{{- range $telemetryName, $telemetryParams := .Values.telemetry }}
-  {{- if index $telemetryParams "enabled" }}
-    {{- $config := include (printf "opentelemetry-collector.%sPipelineConfig" $telemetryName) $ | fromYaml | mustMergeOverwrite $config }}
-  {{- end }}
+{{- define "opentelemetry-collector.metrics.enabled" -}}
+{{- or .Values.configOverride.service.pipelines.metrics  .Values.agentCollectorconfigOverride.service.pipelines.metrics .Values.standaloneCollector.configOverride.service.pipelines.metrics }}
 {{- end }}
-{{- $config | toYaml }}
+
+{{/*
+*/}}
+{{- define "opentelemetry-collector.traces.enabled" -}}
+{{- or .Values.configOverride.service.pipelines.traces  .Values.agentCollectorconfigOverride.service.pipelines.traces .Values.standaloneCollector.configOverride.service.pipelines.traces }}
+{{- end }}
+
+{{/*
+Build base config with top level override
+*/}}
+{{- define "opentelemetry-collector.overridenConfig" -}}
+{{- $source := .Values.configOverride  -}}
+{{- $config := include "opentelemetry-collector.baseConfig" . | fromYaml  -}}
+{{- if "opentelemetry-collector.metrics.enabled" }}
+{{- $config := include (printf "opentelemetry-collector.metricsPipelineConfig" ) $ | fromYaml | mustMergeOverwrite $config }}
+{{- end }}
+{{- if "opentelemetry-collector.traces.enabled" }}
+{{- $config := include (printf "opentelemetry-collector.tracesPipelineConfig" ) $ | fromYaml | mustMergeOverwrite $config }}
+{{- end }}
+{{- $source | mergeOverwrite $config  | toYaml }}
```

Comment:
seems like `$source` defined only once, why don't set it right here?

also can we use `mustMergeOverwrite` to catch errors and for consistency?